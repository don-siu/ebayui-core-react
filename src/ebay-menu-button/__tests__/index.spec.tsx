import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import initStoryshots from '@storybook/addon-storyshots'
import { EbayMenuButton, EbayMenuButtonItem } from '..'

initStoryshots({
    config: ({ configure }) => configure(() => require('./index.stories.tsx'), module)
})

describe('<EbayMenuButton>', () => {
    describe('on button click', () => {
        let spy
        beforeEach(() => {
            spy = jest.fn()
        })
        it('should fire onExpand event', () => {
            const wrapper = render(
                <EbayMenuButton onExpand={spy}>
                    <EbayMenuButtonItem />
                </EbayMenuButton>
            )
            fireEvent.click(wrapper.container.querySelector('button'))

            expect(spy).toBeCalled()
        })
        it('should fire onCollapse event', () => {
            const wrapper = render(
                <EbayMenuButton onCollapse={spy}>
                    <EbayMenuButtonItem />
                </EbayMenuButton>
            )

            const button = wrapper.container.querySelector('button')
            fireEvent.click(button)
            fireEvent.click(button)

            expect(spy).toBeCalled()
        })
        it('should fire onChange event', () => {
            const wrapper = render(
                <EbayMenuButton type="radio" onChange={spy}>
                    <EbayMenuButtonItem />
                    <EbayMenuButtonItem />
                </EbayMenuButton>
            )

            const button = wrapper.container.querySelector('button')
            fireEvent.click(button)
            const item = screen.getAllByRole('menuitem')[1]
            fireEvent.click(item)

            expect(spy).toBeCalledWith(1, expect.anything())
        })
        it('should fire onSelect event', () => {
            const wrapper = render(
                <EbayMenuButton type="checkbox" onChange={spy}>
                    <EbayMenuButtonItem />
                    <EbayMenuButtonItem />
                </EbayMenuButton>
            )

            const button = wrapper.container.querySelector('button')
            fireEvent.click(button)
            const item = screen.getAllByRole('menuitem')[1]
            fireEvent.click(item)

            expect(spy).toBeCalledWith(1, expect.anything())
        })
    })
})
