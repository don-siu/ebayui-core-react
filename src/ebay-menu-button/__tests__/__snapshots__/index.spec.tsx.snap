// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ebay-menu-button Borderless 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--borderless"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        eBay Menu without borders!
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
  <span
    className="menu-button__menu menu"
  >
    <div
      className="menu__items"
      role="menu"
    >
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 1
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 2
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 3
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
    </div>
  </span>
</span>
`;

exports[`Storyshots ebay-menu-button Default 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        eBay Menu
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
  <span
    className="menu-button__menu menu"
  >
    <div
      className="menu__items"
      role="menu"
    >
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 1 that has very long text
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 2
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 3
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
    </div>
  </span>
</span>
`;

exports[`Storyshots ebay-menu-button Fixed Width 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        Menu has a button width
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
  <span
    className="menu-button__menu menu-button__menu--fix-width menu"
  >
    <div
      className="menu__items"
      role="menu"
    >
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 1 that has very long text
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 2
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 3
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
    </div>
  </span>
</span>
`;

exports[`Storyshots ebay-menu-button Multi-Select Menu Button 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        Filter
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
  <span
    className="menu-button__menu menu"
  >
    <div
      className="menu__items"
      role="menu"
    >
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          Snickers
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          T-Shirts
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          Pants
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
    </div>
  </span>
</span>
`;

exports[`Storyshots ebay-menu-button Reverse (Menu grows to the left) 1`] = `
<span
  className="menu-button"
  style={
    Object {
      "marginLeft": "100px",
    }
  }
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        Menu grows to the left
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
  <span
    className="menu-button__menu menu-button__menu--reverse menu"
  >
    <div
      className="menu__items"
      role="menu"
    >
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 1 that has very very long text
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 2
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 3
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
    </div>
  </span>
</span>
`;

exports[`Storyshots ebay-menu-button Single-Select Menu Button 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        Sort order
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
  <span
    className="menu-button__menu menu"
  >
    <div
      className="menu__items"
      role="menu"
    >
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          Date
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          Price
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          Relevance
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
    </div>
  </span>
</span>
`;

exports[`Storyshots ebay-menu-button Variant: Overflow (Icon Button) 1`] = `
<span
  a11y-text="Menu"
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="menu-button__button icon-btn"
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="icon icon--overflow"
      focusable={false}
      height="24px"
      width="24px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <use
        xlinkHref="#icon-overflow"
      />
    </svg>
  </button>
  <span
    className="menu-button__menu menu"
  >
    <div
      className="menu__items"
      role="menu"
    >
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 1
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 2
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 3
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
    </div>
  </span>
</span>
`;

exports[`Storyshots ebay-menu-button With Custom Label 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        <span
          style={
            Object {
              "background": "url(\\"data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='25' height='15' viewBox='0 0 5 3'%3E%3Cpath d='M0 0h5v3H0z'/%3E%3Cpath fill='%23D00' d='M0 1h5v2H0z'/%3E%3Cpath fill='%23FFCE00' d='M0 2h5v1H0z'/%3E%3C/svg%3E\\") no-repeat 50% 50%",
              "display": "inline-block",
              "height": "20px",
              "marginRight": "8px",
              "verticalAlign": "middle",
              "width": "26px",
            }
          }
        />
        Fun with flags!
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
  <span
    className="menu-button__menu menu"
  >
    <div
      className="menu__items"
      role="menu"
    >
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 1
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 2
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 3
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
    </div>
  </span>
</span>
`;

exports[`Storyshots ebay-menu-button With Separator 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      >
        Complex menu
      </span>
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
  <span
    className="menu-button__menu menu"
  >
    <div
      className="menu__items"
      role="menu"
    >
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 1
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 2
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <hr
        checked={false}
        className="menu-button__item menu-button__separator"
        focused={false}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="separator"
        tabIndex={0}
      />
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 3
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 4
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 5
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <hr
        checked={false}
        className="menu-button__item menu-button__separator"
        focused={false}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="separator"
        tabIndex={0}
      />
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          Log in
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
    </div>
  </span>
</span>
`;

exports[`Storyshots ebay-menu-button Without Text 1`] = `
<span
  className="menu-button"
>
  <button
    aria-expanded={false}
    aria-haspopup={true}
    aria-label="eBay Menu without text"
    aria-live={null}
    className="btn menu-button__button btn--secondary"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="btn__cell"
    >
      <span
        className="btn__text"
      />
      <svg
        aria-hidden={true}
        className="icon icon--dropdown"
        focusable={false}
        height="24px"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <use
          xlinkHref="#icon-dropdown"
        />
      </svg>
    </span>
  </button>
  <span
    className="menu-button__menu menu"
  >
    <div
      className="menu__items"
      role="menu"
    >
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 1
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 2
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
      <div
        aria-checked={false}
        aria-hidden={false}
        className="menu-button__item menu__item"
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        <span>
          item 3
        </span>
        <svg
          aria-hidden={true}
          className="icon icon--tick-small"
          focusable={false}
          height="16px"
          width="16px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <use
            xlinkHref="#icon-tick-small"
          />
        </svg>
      </div>
    </div>
  </span>
</span>
`;
